src/components/external-events/ExternalEventForm.svelte

```diff
@@ -0,0 +1,220 @@
+<svelte:options immutable={true} />
+
+<script lang="ts">
+  import { plugins } from '../../stores/plugins';
+  import type { ExternalEventDB } from '../../types/external-event';
+  import type { Property } from '../../types/property';
+  import { classNames } from '../../utilities/generic';
+  import { formatDate } from '../../utilities/time';
+  import Collapse from '../Collapse.svelte';
+  import Input from '../form/Input.svelte';
+  import DatePicker from '../ui/DatePicker/DatePicker.svelte';
+  import Highlight from '../ui/Highlight.svelte';
+  import Properties from './Properties.svelte';
+
+  export let externalEvent: ExternalEventDB;
+  export let showHeader: boolean = true;
+
+  let editable: boolean = false;
+  let formProperties: Property[] = [];
+  $: formProperties = Object.entries(externalEvent.properties).map(e => {
+    return {
+      name: e[0],
+      value: e[1],
+    };
+  });
+  let highlightKeysMap: Record<string, boolean> = {};
+</script>
+
+<div class="external-event-form-container">
+  {#if showHeader}
+    <div class="external-event-header">
+      <div class={classNames('external-event-header-title')}>
+        <div class="external-event-header-title-value st-typography-medium">
+          {externalEvent.pkey.key}
+        </div>
+      </div>
+    </div>
+  {/if}
+
+  <div class="external-event-form">
+    <fieldset>
+      <Collapse title="Definition">
+        <!-- <Highlight highlight={highlightKeysMap.id}>
+          <Input layout="inline">
+            ID
+            <input class="st-input w-100" disabled={!editable} name="id" value={externalEvent.id} />
+          </Input>
+        </Highlight> -->
+
+        <Highlight highlight={highlightKeysMap.type}>
```

Comment:
The `Highlight` wrapper is not needed on this form. We use it in the Activity form so that we can highlight certain fields during plan merge review but that shouldn't be applicable here since external events don't participate in plan merge (afaik?)